### PR TITLE
Automate MauiCompat build, Update MAUI Preview 11

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Bug ###

Automate MauiCompat generation, Update MauiCompat to preview 11

### PR Checklist ###

- [ ] Has a linked Issue, and the Issue has been `approved`
- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Review by brminnick — COMMENTED** [review]
Instead of creating a `MauiCompatExists` variable and ensuring it is `True` before attempting to do steps like `Build Xamarin.CommunityToolkit.MauiCompat.csproj`, should we instead fail the build?

I.e. What if we remove `condition: eq (variables['MauiCompatExists'], True)` on all of the MauiCompat steps and let the build fail when the MauiCompat steps don't work? 

I think letting the build fail will ensure future PRs won't break MauiCompat. Am I understanding this correctly?

**Comment by VladislavAntonyuk** [comment]
I thought MauiCompat remains optional if the script fails, but seems like it checks only for path exist, this validation can be removed

File: MauiCompat.sh [hunk 1]
```diff
@@ -7,8 +7,8 @@ dotnet sln ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat.sln add ./
 dotnet new sln -o ./src/Markup/ -n Xamarin.CommunityToolkit.Markup.MauiCompat
 dotnet sln ./src/Markup/Xamarin.CommunityToolkit.Markup.MauiCompat.sln add ./src/Markup/Xamarin.CommunityToolkit.Markup.MauiCompat/Xamarin.CommunityToolkit.Markup.MauiCompat.csproj
 
-sed -i '' 's/;net6.0-maccatalyst//g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/**/Xamarin.CommunityToolkit.MauiCompat.csproj
-sed -i '' 's/;net6.0-maccatalyst//g' ./src/Markup/Xamarin.CommunityToolkit.Markup.MauiCompat/**/Xamarin.CommunityToolkit.Markup.MauiCompat.csproj
+sed -i '' 's/<ImplicitUsings>enable/<ImplicitUsings>false/g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/**/Xamarin.CommunityToolkit.MauiCompat.csproj
```
Comment (VladislavAntonyuk) [inline MauiCompat.sh:10]: disable implicit usings. it gives us more control

File: MauiCompat.sh [hunk 2]
```diff
@@ -61,8 +61,8 @@ printf > ./src/CommunityToolkit/Directory.build.props "<Project>
     </ItemGroup>
 
     <ItemGroup Condition=\" \$(TargetFramework.Contains('-maccatalyst')) \">
-        <Compile Include=\"**\*.macos.cs\" />
-        <Compile Include=\"**\*.macos.*.cs\" />
+        <Compile Include=\"**\*.ios.cs\" />
```
Comment (VladislavAntonyuk) [inline MauiCompat.sh:64]: maccatalyst uses ios files

File: MauiCompat.sh [hunk 3]
```diff
@@ -276,6 +276,14 @@ sed -i '' 's/ConvertFromInvariantString(string value)/ConvertFrom(System.Compone
 
 sed -i '' 's/ConvertFromInvariantString(string\? value)/ConvertFrom(System.ComponentModel.ITypeDescriptorContext? context, System.Globalization.CultureInfo? culture, object valueObject)/g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/**/**.cs
 
+sed -i '' 's/if (value != null)/if (valueObject is not string value){throw new InvalidOperationException("Only typeof(string) allowed");}if (value != null)/g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/Helpers/SafeAreaTypeConverter.shared.cs
```
Comment (VladislavAntonyuk) [inline MauiCompat.sh:279]: automate converters

File: MauiCompat.sh [hunk 4]
```diff
@@ -493,3 +500,88 @@ sed -i '' 's/Xamarin.Forms.RelativeLayout/Microsoft.Maui.Controls.Compatibility.
...
+			{
+				colorToConvert = Microsoft.Maui.Graphics.Colors.Transparent;
+			}
+
+			return Microsoft.Maui.Controls.Compatibility.Platform.iOS.ColorExtensions.ToUIColor(colorToConvert);
+		}
+	}
+}"
+
+sed -i '' 's/public class/using Xamarin.CommunityToolkit.MauiCompat; public class/g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/Effects/IconTintColor/IconTintColorEffectRouter.ios.cs
```
Comment (VladislavAntonyuk) [inline MauiCompat.sh:546]: automate MauiColor extensions (ToAndroid, ToUIColor)

File: MauiCompat.sh [hunk 5]
```diff
@@ -493,3 +500,88 @@ sed -i '' 's/Xamarin.Forms.RelativeLayout/Microsoft.Maui.Controls.Compatibility.
...
+# AVMediaType
+
+sed -i '' 's/photoOutput.ConnectionFromMediaType(AVMediaType.Video)/photoOutput.ConnectionFromMediaType(new NSString("video"))/g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/**/**.cs
+sed -i '' 's/AVCaptureDevice.DevicesWithMediaType(AVMediaType.Video)/AVCaptureDevice.DevicesWithMediaType("video")/g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/**/**.cs
+sed -i '' 's/AVMediaType.Audio/AVMediaTypes.Audio/g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/**/**.cs
+sed -i '' 's/AVMediaType.Video/AVAuthorizationMediaType.Video/g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/**/**.cs
+
+# Log
+
+sed -i '' 's/using System;/using System;using Microsoft.Extensions.Logging;/g' ./src/CommunityToolkit/Xamarin.CommunityToolkit.MauiCompat/**/**.cs
```
Comment (VladislavAntonyuk) [inline MauiCompat.sh:586]: https://github.com/dotnet/maui/pull/3401

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Core/FileMediaSourceConverter.shared.cs [hunk 6]
```diff
@@ -7,9 +7,11 @@ namespace Xamarin.CommunityToolkit.Core
 	[TypeConversion(typeof(FileMediaSource))]
 	public sealed class FileMediaSourceConverter : TypeConverter
 	{
-		public override object ConvertFromInvariantString(string value) =>
-			value != null
+		public override object ConvertFromInvariantString(string value)
+		{
+			return value != null
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Core/FileMediaSourceConverter.shared.cs:12]: required for simple replacement

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/UriTypeConverter.shared.cs [hunk 7]
```diff
@@ -7,7 +7,9 @@ namespace Xamarin.CommunityToolkit.UI.Views
 	[TypeConversion(typeof(Uri))]
 	public class UriTypeConverter : TypeConverter
 	{
-		public override object? ConvertFromInvariantString(string? value) =>
-			string.IsNullOrWhiteSpace(value) ? null : new Uri(value, UriKind.RelativeOrAbsolute);
+		public override object? ConvertFromInvariantString(string? value)
+		{
+			return string.IsNullOrWhiteSpace(value) ? null : new Uri(value, UriKind.RelativeOrAbsolute);
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/UriTypeConverter.shared.cs:12]: required for simple replacement

**Comment by VladislavAntonyuk** [comment]
I will rework pipeline logic a bit. We need to skip script if MauiCompat exists ( for branch like 1.3.0-MauiCompat)

File: MauiCompat.sh (hunk 2 above)
Comment (VladislavAntonyuk) [reply MauiCompat.sh:64]: windows is not supported, because of transition UWP to WinUI

**Review by brminnick — APPROVED** [review]

**Comment by brminnick** [comment]
Thanks Vlad!!

**Pull request merged**